core/store/src/db.rs

```diff
@@ -415,9 +448,9 @@ impl RocksDB {
     pub fn new<P: AsRef<std::path::Path>>(path: P) -> Result<Self, DBError> {
         let options = rocksdb_options();
         let cf_names: Vec<_> = (0..NUM_COLS).map(|col| format!("col{}", col)).collect();
-        let cf_descriptors = cf_names
-            .iter()
-            .map(|cf_name| ColumnFamilyDescriptor::new(cf_name, rocksdb_column_options()));
+        let cf_descriptors = cf_names.iter().enumerate().map(|(col_index, cf_name)| {
+            ColumnFamilyDescriptor::new(cf_name, rocksdb_column_options(col_index))
```

Comment:
Would you have a native iterator over all the enum variants (see ^^^), you would be able to pass a DBCol value and avoid the use of `usize` casting which creates implicit connections between an index and the DBCol value. We can leverage strong type system to help us avoid possible bugs.